drivers/note/noteram_driver.c

```diff
@@ -57,28 +58,33 @@ struct noteram_info_s
 {
   volatile unsigned int ni_head;
   volatile unsigned int ni_tail;
+  volatile unsigned int ni_read;
```

Comment:
why we need ni_tail and ni_read two location?

Reply:
Same as https://github.com/apache/incubator-nuttx/pull/1377#discussion_r493667049
I have wanted to introduce non-destructive read to /dev/note.
It is useful for example no overwrite mode in on-boot instrumentation. After boot, the instrumentation data while boot is kept in the circular buffer and never lost until new instrumentation is started even when we read the data.